recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py

```diff
@@ -0,0 +1,618 @@
+    then we will not convert them again (so you may need to delete them
+    by hand if you, at any point, change your ARPA model).
+
+    Arguments
+    ---------
+        arpa_dir: str
+            Path to the directory containing the ARPA LM (we expect a file named
+            3-gram.pruned.1e-7.arpa to exist, and if `convert_4gram` is True,
+            then "4-gram.arpa" should also exist).
+        output_dir: str
+            Path to the directory where the FSTs will be saved.
+        words_txt: str
+            Path to the words.txt file created by prepare_lang.
+        disambig_symbol: str
+            The disambiguation symbol to use.
+        convert_4gram: bool
+            If True, then we will convert the 4-gram ARPA LM to
+            FST. Otherwise, we will only convert the 3-gram ARPA LM to FST.
+        trigram_name: str
+            The name of the 3-gram ARPA LM file.
+        fourgram_name: str
+            The name of the 4-gram ARPA LM file.
+    
+    Raises
+    ---------
+        ImportError: If kaldilm is not installed.
+    """
+    assert arpa_dir.is_dir()
+    assert output_dir.is_dir()
+    try:
+        from kaldilm.arpa2fst import arpa2fst
+    except ImportError:
+        # This error will occur when there is fst LM in the provided lm_dir
+        # and we are trying to create it by converting an ARPA LM to FST.
+        # For this, we need to install kaldilm.
+        raise ImportError(
+            "Optional dependencies must be installed to use kaldilm.\n"
+            "Install using `pip install kaldilm`."
+        )
+    def _arpa_to_fst_single(arpa_path: Path, out_fst_path: Path, max_order: int):
+        """Convert a single ARPA LM to FST.
+        
+        Arguments
+        ---------
+            arpa_path: str
+                Path to the ARPA LM file.
+            out_fst_path: str
+                Path to the output FST file.
+            max_order: int
+                The maximum order of the ARPA LM.
+        """
+        if out_fst_path.exists():
+            return
+        if not arpa_path.exists():
+            raise FileNotFoundError(f"{arpa_path} not found while trying to create"
+                                    f" the {max_order} FST.")
+        try:
+            s = arpa2fst(
+                input_arpa=str(arpa_path),
+                disambig_symbol=disambig_symbol,
+                read_symbol_table=str(words_txt),
+                max_order=max_order,
+            )
+        except Exception as e:
+            logger.info(f"Failed to create {max_order}-gram FST from input={arpa_path}"
+                        f", disambig_symbol={disambig_symbol},"
+                        f" read_symbol_table={words_txt}")
+            raise e
+        logger.info(f"Writing {out_fst_path}")
+        with open(out_fst_path, "w") as f:
+            f.write(s)
+    arpa_path = arpa_dir / trigram_name
+    fst_path = output_dir / "G_3_gram.fst.txt"
```

Comment:
do you think we should turn `"G_3_gram.fst.txt` as an input variable to the function ? Maybe defined by the user? (same for 4gram)